src/main/java/com/igot/cb/contentpartner/service/impl/ContentPartnerRegistrationServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+import com.igot.cb.pores.elasticsearch.service.EsUtilService;
+import com.igot.cb.pores.util.ApiResponse;
+import com.igot.cb.pores.util.CbServerProperties;
+import com.igot.cb.pores.util.Constants;
+import com.igot.cb.pores.util.PayloadValidation;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class ContentPartnerRegistrationServiceImpl implements ContentPartnerRegistrationService {
+    @Autowired
+    private PayloadValidation payloadValidation;
+    @Autowired
+    private ContentPartnerRegistrationRepository registrationRepository;
+    @Autowired
+    private CacheService cacheService;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private CbServerProperties cbServerProperties;
+    @Autowired
+    private EsUtilService esUtilService;
+
+
+    @Override
+    public ApiResponse createOrUpdate(JsonNode partnerDetails) {
+        log.info("ContentPartnerServiceImpl::createOrUpdate:inside");
+        ApiResponse response = new ApiResponse();
+        try {
+            if (partnerDetails.get(Constants.ID) == null) {
+                response = createContentPartnerRegistration(partnerDetails);
+            } else {
+                response = updateContentPartner(partnerDetails);
+            }
+            return response;
+        } catch (Exception e) {
+            response.getParams().setErrMsg(e.getMessage());
+            response.getParams().setStatus(Constants.FAILED);
+            response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+            return response;
+        }
+    }
+
+    private ApiResponse createContentPartnerRegistration(JsonNode registrationDetails) {
+        log.info("ContentPartnerRegistrationServiceImpl::createContentPartnerRegistration");
+
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_PARTNER_CREATE);
+        Timestamp currentTime = new Timestamp(System.currentTimeMillis());
+        payloadValidation.validatePayload(Constants.PAYLOAD_VALIDATION_FILE_CONTENT_PARTNER_REGISTRATION, registrationDetails);
+        String organizationName = registrationDetails.path("contentPartnerName").asText("");
+        String email = registrationDetails.path("email").asText("");
+
+        Optional<ContentPartnerRegistrationEntity> existingByOrgName =
+                registrationRepository.findByContentPartnerOrganizationName(organizationName);
+
+        if (existingByOrgName.isPresent()) {
+            response.getParams().setErrMsg("Organization Name already registered");
+            response.getParams().setStatus(Constants.FAILED);
+            response.setResponseCode(HttpStatus.BAD_REQUEST);
```

Comment:
these 3 lines are repeated. check the other repos.. we will have utility method to update the error response